### PR TITLE
Feature/optimize sql requests

### PR DESCRIPTION
**Pull request by Lex-23**

### CODE REVIEW

**Comment by Lex-23** [comment]
[Upgrade/modify SQL requests](https://trello.com/c/gWS7tZv5/14-upgrade-modify-sql-requests)

File: cryptostock/account/tests/test_jwt_auth.py [hunk 1]
```diff
@@ -12,6 +12,8 @@ def test_jwt_auth(user_one, auth_broker):
     assert response.status_code == 200
     assert "refresh" in response.json().keys() and "access" in response.json().keys()
     assert token["user_role"] == "broker"
+    assert response.json()["refresh"] != response.json()["access"]
+    assert len(response.json().keys()) == 2
```
Comment (andreyPromaster) [inline cryptostock/account/tests/test_jwt_auth.py:16]:
```suggestion
    assert len(response.json()) == 2
```

File: cryptostock/utils/services.py [hunk 2]
```diff
@@ -83,8 +83,24 @@ def offer_flow(offer_count, request, deal) -> dict:
...
-    elif self.request.auth["user_role"] == "broker":
-        return Offer.objects.filter(deal__broker__id=self.request.user.account.id)
+def _get_offers(request):
+    if request.auth["user_role"] == "client":
+        return Offer.objects.filter(client_id=request.user.account.client.id)
+    elif request.auth["user_role"] == "broker":
+        return Offer.objects.filter(deal__broker__id=request.user.account.broker.id)
+
+
+def optimized_get_offers(request):
```
Comment (andreyPromaster) [inline cryptostock/utils/services.py:93]:
```suggestion
def get_offers_with_related_items(request):
```

Comment (Lex-23) [reply cryptostock/utils/services.py]: fixed

File: cryptostock/account/tests/test_delete_salesdashboard.py [hunk 3]
```diff
@@ -2,15 +2,20 @@
...
 
 
 def test_delete_sales_dashboard(auth_broker, broker_account):
     sale = SalesDashboardFactory(broker=broker_account)
 
-    response = auth_broker.delete(f"/api/salesdashboard/{sale.id}/")
+    with CaptureQueriesContext(connection) as query_context:
+        response = auth_broker.delete(f"/api/salesdashboard/{sale.id}/")
     with pytest.raises(ObjectDoesNotExist, match=r".* does not exist."):
         SalesDashboard.objects.get(id=sale.id)
```
[the remark this answers is not in the document]
Comment (Lex-23) [reply cryptostock/account/tests/test_delete_salesdashboard.py]: agree.

File: cryptostock/account/tests/test_get_offer.py [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 
 from cryptostock.settings import REST_FRAMEWORK as DRF
 
 
 def test_client_get_offer(auth_client, client_account):
     offer = OfferFactory(client=client_account)
 
-    response = auth_client.get(f"/api/offer/{offer.id}/")
+    with CaptureQueriesContext(connection) as query_context:
+        response = auth_client.get(f"/api/offer/{offer.id}/")
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_offer.py:12]: We're agree on moving db_calls into separate test, isn't it?

Comment (Lex-23) [reply cryptostock/account/tests/test_get_offer.py:12]: in all tests?

Comment (Lex-23) [reply cryptostock/account/tests/test_get_offer.py]: got it

File: cryptostock/account/tests/test_delete_salesdashboard.py [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
+
+
+def test_delete_sales_dashboard_db_calls(auth_broker, broker_account):
+    sale = SalesDashboardFactory(broker=broker_account)
+
+    with CaptureQueriesContext(connection) as query_context:
+        response = auth_broker.delete(f"/api/salesdashboard/{sale.id}/")
+
     assert response.status_code == 204
+    assert len(query_context) == 7
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_delete_salesdashboard.py:28]: Why 7 ?

**Review by vadim-denisov — APPROVED** [review]

**Pull request merged**